### DISCUSSION TITLE
How to I apply text-overflow to button text?

### DISCUSSION THREAD

**Discussion by datvm** [material-components/material-web#5246] [Q&A] [answered]
I have a button with unknown content and I want to use `text-overflow: ellipsis`. I managed to keep the button size within desired size but the text still shows up and overflows:

![image](https://github.com/material-components/material-web/assets/6388546/f077fdc9-c8d3-46b4-887d-7935018bfe61)

```css
    .button-text {
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }
```

[Lit Playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQgXCJAbWF0ZXJpYWwvd2ViL2FsbFwiIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHk-XG4gIDxzdHlsZT5cbiAgICAuY29udGFpbmVyIHtcbiAgICAgIHdpZHRoOiAxNTBweDtcbiAgICB9XG4gICAgXG4gICAgbWQtZWxldmF0ZWQtYnV0dG9uIHtcbiAgICAgIG1heC13aWR0aDogMTAwJTtcbiAgICB9XG4gICAgXG4gICAgLmJ1dHRvbi10ZXh0IHtcbiAgICAgIG92ZXJmbG93OiBoaWRkZW47XG4gICAgICB3aGl0ZS1zcGFjZTogbm93cmFwO1xuICAgICAgdGV4dC1vdmVyZmxvdzogZWxsaXBzaXM7XG4gICAgfVxuICA8L3N0eWxlPlxuICBcbiAgPGRpdiBjbGFzcz1cImNvbnRhaW5lclwiPlxuICAgIDxtZC1lbGV2YXRlZC1idXR0b24-XG4gICAgICA8ZGl2IGNsYXNzPVwiYnV0dG9uLXRleHRcIj5Tb21lIGxvbmcgdGV4dCBjb250ZW50IHRoYXQgc2hvdWxkIGJlIGNvbGxhcHNlZDwvZGl2PlxuICAgIDwvbWQtZWxldmF0ZWQtYnV0dG9uPlxuICA8L2Rpdj5cbjwvYm9keT5cbiJ9LHsibmFtZSI6InBhY2thZ2UuanNvbiIsImNvbnRlbnQiOiJ7XG4gIFwiZGVwZW5kZW5jaWVzXCI6IHtcbiAgICBcImxpdFwiOiBcIl4zLjAuMFwiLFxuICAgIFwiQGxpdC9yZWFjdGl2ZS1lbGVtZW50XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJsaXQtZWxlbWVudFwiOiBcIl40LjAuMFwiLFxuICAgIFwibGl0LWh0bWxcIjogXCJeMy4wLjBcIlxuICB9XG59IiwiaGlkZGVuIjp0cnVlfV0)

In Devtools, if I add `max-width: 100%` to its parent (`.button__label`), it works but I don't know how I can do that without DevTools. Please give me some direction. Thanks.

**Comment by CornerSyrup** [comment]
It doesn't seems possible, for the width of `span.button__label` doesn't limited by it's container, and there's no token to customize the `overflow` or `width`/`max-width`.

I've thought of `::part` pseudo-selector but MWC can't be customized via `part`.

my experiment:
![image](https://github.com/material-components/material-web/assets/65886108/1d482d77-0372-46f6-8be3-89f64f98eb76)

**Comment by asyncliz** [comment, marked as answer]
Can you try with the nightly version? We added this as a feature.

**Reply by datvm** [reply]
Thanks, it works in the latest version :) 

![image](https://github.com/material-components/material-web/assets/6388546/6f02ed01-fb0b-4d84-a055-bf22bb20b0a2)

P.s: is there any way to test current Github code with Lit Playground? Currently I have to pull the Github code to the local machine to test it.

**Reply by asyncliz** [reply]
Not right now I'm afraid, you'd need to set up a manual CDN

```ts
// import '@material/web/all.js';
import 'https://esm.run/@material/web@nightly/all.js';
```
